presto-common/src/test/java/com/facebook/presto/common/type/TestTimeZoneKey.java

```diff
@@ -46,89 +46,87 @@ public void testUTC()
-        assertSame(TimeZoneKey.getTimeZoneKey("zulu"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("ZULU"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("UT"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("UCT"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("Universal"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT0"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT+0"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT-0"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT+00:00"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT-00:00"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("+00:00"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("-00:00"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("etc/utc"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("etc/gmt"), UTC_KEY);
+        assertTimeZoneNotSupported("Z");
+        assertTimeZoneNotSupported("Zulu");
+        assertTimeZoneNotSupported("zulu");
+        assertTimeZoneNotSupported("ZULU");
+        assertTimeZoneNotSupported("UT");
+        assertTimeZoneNotSupported("UCT");
+        assertTimeZoneNotSupported("Universal");
+        assertTimeZoneNotSupported("GMT");
+        assertTimeZoneNotSupported("GMT0");
+        assertTimeZoneNotSupported("GMT+0");
+        assertTimeZoneNotSupported("GMT-0");
+        assertTimeZoneNotSupported("GMT+00:00");
+        assertTimeZoneNotSupported("GMT-00:00");
+        assertTimeZoneNotSupported("+00:00");
+        assertTimeZoneNotSupported("-00:00");
+        assertTimeZoneNotSupported("etc/utc");
+        assertTimeZoneNotSupported("etc/gmt");
         assertSame(TimeZoneKey.getTimeZoneKey("etc/gmt+0"), UTC_KEY);
         assertSame(TimeZoneKey.getTimeZoneKey("etc/gmt+00:00"), UTC_KEY);
         assertSame(TimeZoneKey.getTimeZoneKey("etc/gmt-00:00"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("etc/ut"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("etc/UT"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("etc/UCT"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("etc/Universal"), UTC_KEY);
+        assertTimeZoneNotSupported("etc/ut");
+        assertTimeZoneNotSupported("etc/UT");
+        assertTimeZoneNotSupported("etc/UCT");
+        assertTimeZoneNotSupported("etc/Universal");
     }
 
     @Test
     public void testHourOffsetZone()
     {
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT0"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT+0"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT-0"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT+0"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT-0"), UTC_KEY);
+        assertTimeZoneNotSupported("GMT0");
+        assertTimeZoneNotSupported("GMT+0");
```

Comment:
This should be supporte.  https://en.wikipedia.org/wiki/List_of_tz_database_time_zones